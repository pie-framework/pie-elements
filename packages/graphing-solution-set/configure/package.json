{
  "name": "@pie-element/graphing-solution-set-configure",
  "private": true,
  "version": "2.3.6",
  "description": "",
  "main": "lib/index.js",
  "module": "src/index.js",
  "author": "",
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@pie-framework/pie-configure-events": "^1.3.0",
    "@pie-lib/config-ui": "^11.18.1",
    "@pie-lib/editable-html": "^11.10.1",
    "@pie-lib/graphing-solution-set": "^2.23.1",
    "@pie-lib/math-rendering": "^3.11.1",
    "classnames": "^2.2.5",
    "debug": "^4.1.1",
    "prop-types": "^15.6.2",
    "react": "^16.8.1",
    "react-dom": "^16.8.1"
  },
  "scripts": {
    "test": "./node_modules/.bin/jest"
  },
  "license": "ISC"
}
