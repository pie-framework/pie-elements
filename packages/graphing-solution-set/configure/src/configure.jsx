import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import { settings, layout, InputContainer } from '@pie-lib/pie-toolbox/config-ui';
import PropTypes from 'prop-types';
import debug from 'debug';
import Typography from '@material-ui/core/Typography';
import { EditableHtml } from '@pie-lib/pie-toolbox/editable-html';
import GraphingConfig from './graphing-config';
import CorrectResponse from './correct-response';
import intersection from 'lodash/intersection';

const { Panel, toggle, radio, checkboxes, textField, dropdown } = settings;
const log = debug('@pie-element:graphing-solution-set:configure');

const styles = (theme) => ({
  title: {
    fontSize: '1.1rem',
    display: 'block',
    marginTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit,
  },
  promptHolder: {
    width: '100%',
    paddingTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2,
    marginTop: theme.spacing.unit * 2,
  },
  description: {
    marginBottom: theme.spacing.unit * 2.5,
  },
});

export class Configure extends React.Component {
  static propTypes = {
    onModelChanged: PropTypes.func,
    onConfigurationChanged: PropTypes.func,
    classes: PropTypes.object,
    imageSupport: PropTypes.object,
    uploadSoundSupport: PropTypes.object,
    model: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
  };

  static defaultProps = { classes: {} };

  /*
   * Triggered when configure component is mounted
   * */
  componentDidMount() {
    const { configuration, onModelChanged, model } = this.props;
    const { title, graphDimensions } = configuration || {};
    let { arrows, titleEnabled: showTitle, dimensionsEnabled: showDimensions } = model || {};
    // This is used for offering support for old models which have the property arrows: boolean
    // Same thing is set in the controller: packages/graphing/controller/src/index.js - model
    if (typeof arrows === 'boolean') {
      if (arrows) {
        arrows = { left: true, right: true, up: true, down: true };
      } else {
        arrows = { left: false, right: false, up: false, down: false };
      }
    }
    const titleEnabled = showTitle === undefined || showTitle === null ? title.enabled : showTitle;
    const dimensionsEnabled =
      showDimensions === undefined || showDimensions === null ? graphDimensions.enabled : showDimensions;
    onModelChanged && onModelChanged({ ...model, arrows, titleEnabled, dimensionsEnabled });
  }

  /*
   * Triggered when the rationale is changed
   * @param {string} rationale - The new rationale
   * */
  onRationaleChange = (rationale) => {
    const { onModelChanged, model } = this.props;
    onModelChanged({ ...model, rationale });
  };

  /*
   * Triggered when the prompt is changed
   * @param {string} prompt - The new prompt
   * */
  onPromptChange = (prompt) => {
    const { onModelChanged, model } = this.props;
    onModelChanged({ ...model, prompt });
  };

  /*
   * Triggered when the teacher instructions are changed
   * @param {string} teacherInstructions - The new teacher instructions
   * */
  onTeacherInstructionsChange = (teacherInstructions) => {
    const { onModelChanged, model } = this.props;
    onModelChanged({ ...model, teacherInstructions });
  };

  /*
   * Render the component
   * */
  render() {
    const { classes, model, configuration, onConfigurationChanged, onModelChanged, imageSupport, uploadSoundSupport } =
      this.props;
    const {
      arrows = {},
      authoring = {},
      coordinatesOnHover = {},
      contentDimensions = {},
      gridConfigurations = [],
      graphDimensions = {},
      instruction = {},
      labels = {},
      padding = {},
      prompt = {},
      rationale = {},
      scoringType = {},
      settingsPanelDisabled,
      spellCheck = {},
      studentInstructions = {},
      teacherInstructions = {},
      title = {},
      maxImageWidth = {},
      maxImageHeight = {},
      withRubric = {},
      language = {},
      languageChoices = {},
      mathMlOptions = {},
    } = configuration || {};
    const {
      errors = {},
      extraCSSRules,
      labelsEnabled,
      dimensionsEnabled,
      promptEnabled,
      rationaleEnabled,
      spellCheckEnabled,
      teacherInstructionsEnabled,
      titleEnabled,
    } = model || {};
    const defaultImageMaxWidth = maxImageWidth && maxImageWidth.prompt;
    const defaultImageMaxHeight = maxImageHeight && maxImageHeight.prompt;
    const labelsPlaceholders = {
      top: labels.top,
      right: labels.right,
      bottom: labels.bottom,
      left: labels.left,
    };
    const panelItemType = {
      arrows:
        arrows.settings &&
        checkboxes(arrows.label, {
          left: arrows.left,
          right: arrows.right,
          up: arrows.up,
          down: arrows.down,
        }),
      titleEnabled: title.settings && toggle(title.label),
      padding: padding.settings && toggle(padding.label),
      labelsEnabled: labels.settings && toggle(labels.label),
      'language.enabled': language.settings && toggle(language.label, true),
      language: language.settings && language.enabled && dropdown(languageChoices.label, languageChoices.options),
      dimensionsEnabled: graphDimensions.settings && toggle(graphDimensions.label),
      coordinatesOnHover: coordinatesOnHover.settings && toggle(coordinatesOnHover.label),
    };
    const panelProperties = {
      'authoring.enabled': authoring.settings && toggle(authoring.label, true),
      teacherInstructionsEnabled: teacherInstructions.settings && toggle(teacherInstructions.label),
      studentInstructionsEnabled: studentInstructions.settings && toggle(studentInstructions.label),
      promptEnabled: prompt.settings && toggle(prompt.label),
      rationaleEnabled: rationale.settings && toggle(rationale.label),
      spellCheckEnabled: spellCheck.settings && toggle(spellCheck.label),
      scoringType: scoringType.settings && radio(scoringType.label, ['dichotomous', 'partial scoring']),
      rubricEnabled: withRubric?.settings && toggle(withRubric?.label),
      instruction: instruction.settings && textField(instruction.label),
    };
    return (
      <layout.ConfigLayout
        extraCSSRules={extraCSSRules}
        dimensions={contentDimensions}
        hideSettings={settingsPanelDisabled}
        settings={
          <Panel
            model={model}
            configuration={configuration}
            onChangeModel={onModelChanged}
            onChangeConfiguration={onConfigurationChanged}
            groups={{
              Settings: panelItemType,
              Properties: panelProperties,
            }}
          />
        }
      >
        <Typography component="div" type="body1" className={classes.description}>
          {instruction?.label || ''}
        </Typography>
        {teacherInstructionsEnabled && (
          <InputContainer label={teacherInstructions.label} className={classes.promptHolder}>
            <EditableHtml
              className={classes.prompt}
              markup={model.teacherInstructions || ''}
              onChange={this.onTeacherInstructionsChange}
              imageSupport={imageSupport}
              nonEmpty={false}
              spellCheck={spellCheckEnabled}
              maxImageWidth={(maxImageWidth && maxImageWidth.teacherInstructions) || defaultImageMaxWidth}
              maxImageHeight={(maxImageHeight && maxImageHeight.teacherInstructions) || defaultImageMaxHeight}
              uploadSoundSupport={uploadSoundSupport}
              languageCharactersProps={[{ language: 'spanish' }, { language: 'special' }]}
              mathMlOptions={mathMlOptions}
            />
          </InputContainer>
        )}
        {promptEnabled && (
          <InputContainer label={prompt.label} className={classes.promptHolder}>
            <EditableHtml
              className={classes.prompt}
              markup={model.prompt}
              onChange={this.onPromptChange}
              imageSupport={imageSupport}
              nonEmpty={false}
              spellCheck={spellCheckEnabled}
              disableUnderline
              maxImageWidth={defaultImageMaxWidth}
              maxImageHeight={defaultImageMaxHeight}
              uploadSoundSupport={uploadSoundSupport}
              languageCharactersProps={[{ language: 'spanish' }, { language: 'special' }]}
              mathMlOptions={mathMlOptions}
            />
          </InputContainer>
        )}
        <GraphingConfig
          authoring={authoring}
          gridConfigurations={gridConfigurations}
          graphDimensions={graphDimensions}
          labelsPlaceholders={labelsPlaceholders}
          model={model}
          showLabels={labelsEnabled}
          dimensionsEnabled={dimensionsEnabled}
          showTitle={titleEnabled}
          titlePlaceholder={title.placeholder}
          onChange={this.props.onModelChanged}
          mathMlOptions={mathMlOptions}
        />
        <CorrectResponse
          errors={errors}
          model={model}
          onChange={this.props.onModelChanged}
          mathMlOptions={mathMlOptions}
        />
        {rationaleEnabled && (
          <InputContainer label={rationale.label || 'Rationale'} className={classes.promptHolder}>
            <EditableHtml
              className={classes.prompt}
              markup={model.rationale || ''}
              onChange={this.onRationaleChange}
              imageSupport={imageSupport}
              spellCheck={spellCheckEnabled}
              maxImageWidth={(maxImageWidth && maxImageWidth.rationale) || defaultImageMaxWidth}
              maxImageHeight={(maxImageHeight && maxImageHeight.rationale) || defaultImageMaxHeight}
              uploadSoundSupport={uploadSoundSupport}
              languageCharactersProps={[{ language: 'spanish' }, { language: 'special' }]}
              mathMlOptions={mathMlOptions}
            />
          </InputContainer>
        )}
      </layout.ConfigLayout>
    );
  }
}

export default withStyles(styles)(Configure);
