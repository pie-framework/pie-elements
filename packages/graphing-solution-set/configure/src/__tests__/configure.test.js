import * as React from 'react';
import { shallow } from 'enzyme';
import { Configure } from '../configure';
import defaultValues from '../defaults';

jest.mock('@pie-lib/pie-toolbox/config-ui', () => ({
  InputContainer: (props) => <div>{props.children}</div>,
  InputCheckbox: (props) => <div>{props.children}</div>,
  FeedbackConfig: (props) => <div>{props.children}</div>,
  layout: {
    ConfigLayout: (props) => <div>{props.children}</div>,
  },
  settings: {
    Panel: (props) => <div onChange={props.onChange} />,
    toggle: jest.fn(),
    radio: jest.fn(),
    numberFields: jest.fn(),
    checkboxes: jest.fn(),
    textField: jest.fn(),
  },
}));

jest.mock('@pie-lib/pie-toolbox/graphing-solution-set', () => ({
  GraphContainer: (props) => <div>{props.children}</div>,
  tools: {
    polygon: () => ({
      Component: () => <div />,
      type: 'polygon',
    }),
    line: () => ({
      Component: () => <div />,
      type: 'line',
    }),
  },
}));

describe('Configure', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = (props) => {
      const configureProps = { ...defaultValues, ...props };

      return shallow(<Configure {...configureProps} />);
    };
  });

  describe('renders', () => {
    it('snapshot', () => {
      expect(wrapper()).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    it('updates rationale', () => {
      const onModelChanged = jest.fn();
      const component = wrapper({ onModelChanged });

      component.instance().onRationaleChange('New Rationale');

      expect(onModelChanged).toBeCalledWith(
        expect.objectContaining({
          ...defaultValues.model,
          rationale: 'New Rationale',
        }),
      );
    });

    it('updates prompt', () => {
      const onModelChanged = jest.fn();
      const component = wrapper({ onModelChanged });

      component.instance().onPromptChange('New Prompt');

      expect(onModelChanged).toBeCalledWith(
        expect.objectContaining({
          ...defaultValues.model,
          prompt: 'New Prompt',
        }),
      );
    });

    it('updates teacher instructions', () => {
      const onModelChanged = jest.fn();
      const component = wrapper({ onModelChanged });

      component.instance().onTeacherInstructionsChange('New Teacher Instructions');

      expect(onModelChanged).toBeCalledWith(
        expect.objectContaining({
          ...defaultValues.model,
          teacherInstructions: 'New Teacher Instructions',
        }),
      );
    });
  });
});
