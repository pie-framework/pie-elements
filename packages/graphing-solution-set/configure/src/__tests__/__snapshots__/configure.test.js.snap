// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configure renders snapshot 1`] = `
<ConfigLayout
  dimensions={Object {}}
  hideSettings={false}
  settings={
    <Panel
      configuration={
        Object {
          "arrows": Object {
            "down": Object {
              "label": "down",
            },
            "label": "Include Arrows",
            "left": Object {
              "label": "left",
            },
            "right": Object {
              "label": "right",
            },
            "settings": true,
            "up": Object {
              "label": "up",
            },
          },
          "authoring": Object {
            "axisLabel": Object {
              "enabled": true,
              "label": "Label",
            },
            "enabled": true,
            "includeAxesEnabled": false,
            "label": "Customize Grid Setup",
            "labelStep": Object {
              "enabled": true,
              "label": "Label Interval",
            },
            "max": Object {
              "enabled": true,
              "label": "Max Value",
            },
            "min": Object {
              "enabled": true,
              "label": "Min Value",
            },
            "settings": false,
            "standardGridEnabled": true,
            "step": Object {
              "enabled": true,
              "label": "Grid Interval",
            },
          },
          "coordinatesOnHover": Object {
            "label": "Coordinates on Hover",
            "settings": true,
          },
          "graphDimensions": Object {
            "enabled": true,
            "label": "Graph Dimensions",
            "max": 800,
            "min": 150,
            "settings": false,
            "step": 20,
          },
          "gridConfigurations": Array [
            Object {
              "arrows": Object {
                "down": true,
                "left": true,
                "right": true,
                "up": true,
              },
              "domain": Object {
                "axisLabel": "<i>x</i>",
                "labelStep": 1,
                "max": 10,
                "min": -10,
                "padding": 0,
                "step": 1,
              },
              "graph": Object {
                "height": 480,
                "width": 480,
              },
              "includeAxes": true,
              "label": "4-quadrant coordinate grid, -10 to 10",
              "labels": Object {
                "bottom": "",
                "left": "",
                "right": "",
                "top": "",
              },
              "padding": true,
              "range": Object {
                "axisLabel": "<i>y</i>",
                "labelStep": 1,
                "max": 10,
                "min": -10,
                "padding": 0,
                "step": 1,
              },
              "standardGrid": true,
              "title": "",
            },
            Object {
              "arrows": Object {
                "down": false,
                "left": false,
                "right": true,
                "up": true,
              },
              "domain": Object {
                "axisLabel": "<i>x</i>",
                "labelStep": 1,
                "max": 10,
                "min": 0,
                "padding": 0,
                "step": 1,
              },
              "graph": Object {
                "height": 480,
                "width": 480,
              },
              "includeAxes": true,
              "label": "0 to 10 on both axes",
              "labels": Object {
                "bottom": "",
                "left": "",
                "right": "",
                "top": "",
              },
              "padding": true,
              "range": Object {
                "axisLabel": "<i>y</i>",
                "labelStep": 1,
                "max": 10,
                "min": 0,
                "padding": 0,
                "step": 1,
              },
              "standardGrid": true,
              "title": "",
            },
            Object {
              "arrows": Object {
                "down": false,
                "left": false,
                "right": true,
                "up": true,
              },
              "domain": Object {
                "axisLabel": "<i>x</i>",
                "labelStep": 1,
                "max": 20,
                "min": 0,
                "padding": 0,
                "step": 1,
              },
              "graph": Object {
                "height": 480,
                "width": 480,
              },
              "includeAxes": true,
              "label": "0 to 20 on both axes",
              "labels": Object {
                "bottom": "",
                "left": "",
                "right": "",
                "top": "",
              },
              "padding": true,
              "range": Object {
                "axisLabel": "<i>y</i>",
                "labelStep": 1,
                "max": 20,
                "min": 0,
                "padding": 0,
                "step": 1,
              },
              "standardGrid": true,
              "title": "",
            },
            Object {
              "arrows": Object {
                "down": false,
                "left": false,
                "right": true,
                "up": true,
              },
              "domain": Object {
                "axisLabel": "<i>t</i>",
                "labelStep": 2,
                "max": 30,
                "min": 0,
                "padding": 0,
                "step": 1,
              },
              "graph": Object {
                "height": 480,
                "width": 480,
              },
              "includeAxes": true,
              "label": "Sample Data Graph",
              "labels": Object {
                "bottom": "Time (seconds)",
                "left": "Distance (meters)",
                "right": "",
                "top": "",
              },
              "padding": true,
              "range": Object {
                "axisLabel": "<i>d</i>",
                "labelStep": 10,
                "max": 80,
                "min": 0,
                "padding": 0,
                "step": 5,
              },
              "standardGrid": false,
              "title": "Distance as a function of time",
            },
          ],
          "instruction": Object {
            "label": "Graphing Solution Set items involve plotting one or two lines on a graph and then selecting the correct area that represents the solution set. To create one, first configure the grid, then select if one or two lines will be offered and if the lines should be solid or dashed. After defining the line(s), define the correct answer by clicking the area that represents the solution set.",
            "settings": false,
          },
          "labels": Object {
            "bottom": "Click here to add a bottom label",
            "enabled": true,
            "label": "Graph Labels",
            "left": "Click here to add a left label",
            "right": "Click here to add a right label",
            "settings": true,
            "top": "Click here to add a top label",
          },
          "language": Object {
            "enabled": false,
            "label": "Specify Language",
            "settings": false,
          },
          "languageChoices": Object {
            "label": "Language Choices",
            "options": Array [],
          },
          "mathMlOptions": Object {
            "mmlEditing": false,
            "mmlOutput": false,
          },
          "maxImageHeight": Object {
            "prompt": 300,
            "rationale": 300,
            "teacherInstructions": 300,
          },
          "maxImageWidth": Object {
            "prompt": 300,
            "rationale": 300,
            "teacherInstructions": 300,
          },
          "padding": Object {
            "label": "Padding",
            "settings": false,
          },
          "prompt": Object {
            "label": "Item Stem",
            "settings": true,
          },
          "rationale": Object {
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "settingsPanelDisabled": false,
          "spellCheck": Object {
            "enabled": true,
            "label": "Spellcheck",
            "settings": false,
          },
          "studentInstructions": Object {
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "label": "Teacher Instructions",
            "settings": true,
          },
          "title": Object {
            "enabled": true,
            "label": "Graph Title",
            "placeholder": "Click here to add a title for this graph",
            "settings": true,
          },
          "withRubric": Object {
            "label": "Add Rubric",
            "settings": false,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "authoring.enabled": false,
            "instruction": false,
            "promptEnabled": undefined,
            "rationaleEnabled": undefined,
            "rubricEnabled": false,
            "scoringType": false,
            "spellCheckEnabled": false,
            "studentInstructionsEnabled": false,
            "teacherInstructionsEnabled": undefined,
          },
          "Settings": Object {
            "arrows": undefined,
            "coordinatesOnHover": undefined,
            "dimensionsEnabled": false,
            "labelsEnabled": undefined,
            "language": false,
            "language.enabled": false,
            "padding": false,
            "titleEnabled": undefined,
          },
        }
      }
      model={
        Object {
          "answers": Object {
            "correctAnswer": Object {
              "marks": Array [],
              "name": "Correct Answer",
            },
          },
          "arrows": Object {
            "down": true,
            "left": true,
            "right": true,
            "up": true,
          },
          "coordinatesOnHover": false,
          "defaultGridConfiguration": 0,
          "domain": Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "step": 1,
          },
          "graph": Object {
            "height": 500,
            "width": 500,
          },
          "gssLineData": Object {
            "lineA": Object {
              "lineType": "Solid",
            },
            "numberOfLines": 1,
            "sections": Array [],
            "selectedTool": "lineA",
          },
          "includeAxes": true,
          "labels": Object {},
          "labelsEnabled": true,
          "marks": Array [],
          "padding": true,
          "prompt": "",
          "promptEnabled": true,
          "range": Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "step": 1,
          },
          "rationale": "",
          "rationaleEnabled": true,
          "standardGrid": false,
          "studentInstructionsEnabled": true,
          "teacherInstructionsEnabled": true,
          "title": "",
        }
      }
    />
  }
>
  <WithStyles(Typography)
    component="div"
    type="body1"
  >
    Graphing Solution Set items involve plotting one or two lines on a graph and then selecting the correct area that represents the solution set. To create one, first configure the grid, then select if one or two lines will be offered and if the lines should be solid or dashed. After defining the line(s), define the correct answer by clicking the area that represents the solution set.
  </WithStyles(Typography)>
  <InputContainer
    label="Teacher Instructions"
  >
    <EditableHtml
      allowValidation={false}
      languageCharactersProps={
        Array [
          Object {
            "language": "spanish",
          },
          Object {
            "language": "special",
          },
        ]
      }
      markup=""
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={300}
      maxImageWidth={300}
      nonEmpty={false}
      onChange={[Function]}
      onDone={[Function]}
    />
  </InputContainer>
  <InputContainer
    label="Item Stem"
  >
    <EditableHtml
      allowValidation={false}
      disableUnderline={true}
      languageCharactersProps={
        Array [
          Object {
            "language": "spanish",
          },
          Object {
            "language": "special",
          },
        ]
      }
      markup=""
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={300}
      maxImageWidth={300}
      nonEmpty={false}
      onChange={[Function]}
      onDone={[Function]}
    />
  </InputContainer>
  <WithStyles(GraphingConfig)
    authoring={
      Object {
        "axisLabel": Object {
          "enabled": true,
          "label": "Label",
        },
        "enabled": true,
        "includeAxesEnabled": false,
        "label": "Customize Grid Setup",
        "labelStep": Object {
          "enabled": true,
          "label": "Label Interval",
        },
        "max": Object {
          "enabled": true,
          "label": "Max Value",
        },
        "min": Object {
          "enabled": true,
          "label": "Min Value",
        },
        "settings": false,
        "standardGridEnabled": true,
        "step": Object {
          "enabled": true,
          "label": "Grid Interval",
        },
      }
    }
    graphDimensions={
      Object {
        "enabled": true,
        "label": "Graph Dimensions",
        "max": 800,
        "min": 150,
        "settings": false,
        "step": 20,
      }
    }
    gridConfigurations={
      Array [
        Object {
          "arrows": Object {
            "down": true,
            "left": true,
            "right": true,
            "up": true,
          },
          "domain": Object {
            "axisLabel": "<i>x</i>",
            "labelStep": 1,
            "max": 10,
            "min": -10,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "includeAxes": true,
          "label": "4-quadrant coordinate grid, -10 to 10",
          "labels": Object {
            "bottom": "",
            "left": "",
            "right": "",
            "top": "",
          },
          "padding": true,
          "range": Object {
            "axisLabel": "<i>y</i>",
            "labelStep": 1,
            "max": 10,
            "min": -10,
            "padding": 0,
            "step": 1,
          },
          "standardGrid": true,
          "title": "",
        },
        Object {
          "arrows": Object {
            "down": false,
            "left": false,
            "right": true,
            "up": true,
          },
          "domain": Object {
            "axisLabel": "<i>x</i>",
            "labelStep": 1,
            "max": 10,
            "min": 0,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "includeAxes": true,
          "label": "0 to 10 on both axes",
          "labels": Object {
            "bottom": "",
            "left": "",
            "right": "",
            "top": "",
          },
          "padding": true,
          "range": Object {
            "axisLabel": "<i>y</i>",
            "labelStep": 1,
            "max": 10,
            "min": 0,
            "padding": 0,
            "step": 1,
          },
          "standardGrid": true,
          "title": "",
        },
        Object {
          "arrows": Object {
            "down": false,
            "left": false,
            "right": true,
            "up": true,
          },
          "domain": Object {
            "axisLabel": "<i>x</i>",
            "labelStep": 1,
            "max": 20,
            "min": 0,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "includeAxes": true,
          "label": "0 to 20 on both axes",
          "labels": Object {
            "bottom": "",
            "left": "",
            "right": "",
            "top": "",
          },
          "padding": true,
          "range": Object {
            "axisLabel": "<i>y</i>",
            "labelStep": 1,
            "max": 20,
            "min": 0,
            "padding": 0,
            "step": 1,
          },
          "standardGrid": true,
          "title": "",
        },
        Object {
          "arrows": Object {
            "down": false,
            "left": false,
            "right": true,
            "up": true,
          },
          "domain": Object {
            "axisLabel": "<i>t</i>",
            "labelStep": 2,
            "max": 30,
            "min": 0,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "includeAxes": true,
          "label": "Sample Data Graph",
          "labels": Object {
            "bottom": "Time (seconds)",
            "left": "Distance (meters)",
            "right": "",
            "top": "",
          },
          "padding": true,
          "range": Object {
            "axisLabel": "<i>d</i>",
            "labelStep": 10,
            "max": 80,
            "min": 0,
            "padding": 0,
            "step": 5,
          },
          "standardGrid": false,
          "title": "Distance as a function of time",
        },
      ]
    }
    labelsPlaceholders={
      Object {
        "bottom": "Click here to add a bottom label",
        "left": "Click here to add a left label",
        "right": "Click here to add a right label",
        "top": "Click here to add a top label",
      }
    }
    mathMlOptions={
      Object {
        "mmlEditing": false,
        "mmlOutput": false,
      }
    }
    model={
      Object {
        "answers": Object {
          "correctAnswer": Object {
            "marks": Array [],
            "name": "Correct Answer",
          },
        },
        "arrows": Object {
          "down": true,
          "left": true,
          "right": true,
          "up": true,
        },
        "coordinatesOnHover": false,
        "defaultGridConfiguration": 0,
        "domain": Object {
          "axisLabel": "x",
          "labelStep": 1,
          "max": 5,
          "min": -5,
          "step": 1,
        },
        "graph": Object {
          "height": 500,
          "width": 500,
        },
        "gssLineData": Object {
          "lineA": Object {
            "lineType": "Solid",
          },
          "numberOfLines": 1,
          "sections": Array [],
          "selectedTool": "lineA",
        },
        "includeAxes": true,
        "labels": Object {},
        "labelsEnabled": true,
        "marks": Array [],
        "padding": true,
        "prompt": "",
        "promptEnabled": true,
        "range": Object {
          "axisLabel": "y",
          "labelStep": 1,
          "max": 5,
          "min": -5,
          "step": 1,
        },
        "rationale": "",
        "rationaleEnabled": true,
        "standardGrid": false,
        "studentInstructionsEnabled": true,
        "teacherInstructionsEnabled": true,
        "title": "",
      }
    }
    showLabels={true}
    titlePlaceholder="Click here to add a title for this graph"
  />
  <WithStyles(CorrectResponse)
    errors={Object {}}
    mathMlOptions={
      Object {
        "mmlEditing": false,
        "mmlOutput": false,
      }
    }
    model={
      Object {
        "answers": Object {
          "correctAnswer": Object {
            "marks": Array [],
            "name": "Correct Answer",
          },
        },
        "arrows": Object {
          "down": true,
          "left": true,
          "right": true,
          "up": true,
        },
        "coordinatesOnHover": false,
        "defaultGridConfiguration": 0,
        "domain": Object {
          "axisLabel": "x",
          "labelStep": 1,
          "max": 5,
          "min": -5,
          "step": 1,
        },
        "graph": Object {
          "height": 500,
          "width": 500,
        },
        "gssLineData": Object {
          "lineA": Object {
            "lineType": "Solid",
          },
          "numberOfLines": 1,
          "sections": Array [],
          "selectedTool": "lineA",
        },
        "includeAxes": true,
        "labels": Object {},
        "labelsEnabled": true,
        "marks": Array [],
        "padding": true,
        "prompt": "",
        "promptEnabled": true,
        "range": Object {
          "axisLabel": "y",
          "labelStep": 1,
          "max": 5,
          "min": -5,
          "step": 1,
        },
        "rationale": "",
        "rationaleEnabled": true,
        "standardGrid": false,
        "studentInstructionsEnabled": true,
        "teacherInstructionsEnabled": true,
        "title": "",
      }
    }
  />
  <InputContainer
    label="Rationale"
  >
    <EditableHtml
      allowValidation={false}
      languageCharactersProps={
        Array [
          Object {
            "language": "spanish",
          },
          Object {
            "language": "special",
          },
        ]
      }
      markup=""
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={300}
      maxImageWidth={300}
      onChange={[Function]}
      onDone={[Function]}
    />
  </InputContainer>
</ConfigLayout>
`;
